tools/Google.Cloud.Tools.ReleaseProgressReporter/PullRequestDetails.cs

```diff
@@ -44,14 +43,71 @@ public static PullRequestDetails FromUrl(string url)
 
         if (!match.Success)
         {
-            throw new Exception("Not a PR URL.");
+            throw new ArgumentException("Not a PR URL.");
         }
 
         return new PullRequestDetails(match.Groups["owner"].Value,
             match.Groups["repo"].Value,
             int.Parse(match.Groups["number"].Value));
     }
 
+    public static PullRequestDetails? FromEnvironmentVariable()
+    {
+        var pullRequestUrl = Environment.GetEnvironmentVariable("AUTORELEASE_PR");
+        return string.IsNullOrEmpty(pullRequestUrl) ? null : FromUrl(pullRequestUrl);
+    }
+
+    public static async Task<PullRequestDetails?> FromLocalRepo(GitHubClient client)
+    {
+        Console.WriteLine("Determining PR to report progress on");
+        // Otherwise, use the current local repo to work out where to fetch pull requests from.
+        // We currently assume that we're in a single level subdirectory (so ".." is the repo root) - because that's how
+        // the tool is always used at the moment. If that turns out to be a problem, we can make this smarter.
+        using var localRepo = new LibGit2Sharp.Repository("..");
+        var headCommit = localRepo.Head.Tip.Sha;
+
+        var remote = localRepo.Network.Remotes.FirstOrDefault();
+        if (remote is null)
+        {
+            Console.WriteLine("Local git repository has no remotes.");
+            return null;
+        }
+
+        var match = Regex.Match(remote.PushUrl, @"https://github\.com/(?<owner>.+?)/(?<repo>.+?)\.git$");
+        if (!match.Success)
+        {
+            Console.WriteLine($"Cannot determine owner and repo from push URL '{remote.PushUrl}'");
+            return null;
+        }
+        string owner = match.Groups["owner"].Value;
+        string repo = match.Groups["repo"].Value;
+
+        Console.WriteLine("Fetching PRs");
+        // We only fetch PRs created in the 28 days, to avoid fetching huge numbers of ancient PRs.
```

Comment:
```suggestion
        // We only fetch PRs created in the last 28 days, to avoid fetching huge numbers of ancient PRs.
```
```suggestion
        // We only fetch PRs created in the 28 days, to avoid fetching huge numbers of ancient PRs.
```

Reply:
Done.